### PR TITLE
Enforce code style.

### LINKED ISSUE #20: Enforce code style
The repo borrows the coding style (i.e. `.editorconfig`) from the Dapr .NET Client SDK, but doesn't enforce it.  We should do that.

### PR DESCRIPTION
**Pull request by philliphoff**
# Description

This PR runs the formatter in a separate job of the CI workflow.  Also includes minor updates to the source to abide by the formatter.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #20 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -14,6 +14,18 @@ on:
...
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 7.0.x
+      - name: Lint
+        run: dotnet format --verify-no-changes
```
Comment (tmacam) [inline .github/workflows/build.yaml:27]: I know people that hates style enforces. I personally rather have a tool dictate and reformat what the style should be than having to spend time on PRs discussing where brackets should be to improve legibility. And legibility and uniformity matters :)

File: samples/DiscordBindingSample/DiscordBinding.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-using Azure.Storage.Queues;
+﻿using Azure.Storage.Queues;
```
Comment (tmacam) [inline samples/DiscordBindingSample/DiscordBinding.cs:1]: I have no idea what sort of whitespace is here that GitHub "hide whitespace changes" still keeps presenting this diff. Go figure :)

File: src/Dapr.PluggableComponents/AssemblyInfo.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// ------------------------------------------------------------------------
+﻿// ------------------------------------------------------------------------
```
Comment (tmacam) [inline src/Dapr.PluggableComponents/AssemblyInfo.cs:1]: Apparently GitHub can't handle trailing whitespace in files with MS-DOS/Windows line-endings??

**Review by tmacam — APPROVED** [review]
LGTM

File: .github/workflows/build.yaml (hunk 1 above)
Comment (philliphoff) [reply .github/workflows/build.yaml:27]: Yes, I'm happy to just have *a* style defined, and I've grown to care less and less about *which* style it happens to be.

**Pull request merged**